### PR TITLE
Dask slurm backend

### PR DESCRIPTION
**Pull request by pavankum**
## Description
Thank you @SimonBoothroyd for pointing in this direction. I added slurm support for dask jobqueue. Replicated DaskPBSBackend() and checked the output of the job script generated using the following code
```
import openff.evaluator
openff.evaluator.__file__

from openff.evaluator import unit
from openff.evaluator.backends.dask import DaskSLURMBackend
from openff.evaluator.backends import QueueWorkerResources

worker_resources = QueueWorkerResources(
    number_of_threads=1,
    number_of_gpus=1,
    preferred_gpu_toolkit=QueueWorkerResources.GPUToolkit.CUDA,
    per_thread_memory_limit= 1 * unit.gigabytes,
    wallclock_time_limit="05:59",
)

# Create the backend object.
setup_script_commands = [
    f"conda activate evaluator",
    f"module load cuda/10.1.234",
]

calculation_backend = DaskSLURMBackend(
    minimum_number_of_workers=1,
    maximum_number_of_workers=2,
    resources_per_worker=worker_resources,
    queue_name="free-gpu",
    setup_script_commands=setup_script_commands,
    extra_script_options=["--account=account-name", "--gres=gpu:V100:4"],
)

calculation_backend.start()
print(calculation_backend.job_script())
```

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by pavankum** [comment]
By mistake the tutorial updates are also mixed in this update, I can revert them back or can keep them. They're cosmetic changes in setting up the colab environment using conda-colab.

File: openff/evaluator/backends/dask.py [hunk 1]
```diff
@@ -101,6 +101,11 @@ def _wrapper(func, queue, args, kwargs):
                             worker_logger.info(
                                 f"LSBJOBID: {os.environ.get('LSB_JOBID')}"
                             )
+                        elif os.environ.get("SLURM_JOBID") is not None:
```
Comment (SimonBoothroyd) [inline openff/evaluator/backends/dask.py:104]:
```suggestion
                        elif os.environ.get("SLURM_JOB_ID") is not None:
```
I think `SLURM_JOB_ID` is the newer variable name?

File: openff/evaluator/backends/dask.py [hunk 2]
```diff
@@ -695,6 +700,87 @@ def _get_cluster_class(self):
...
+        setup_script_commands=None,
+        extra_script_options=None,
+        adaptive_interval="10000ms",
+        disable_nanny_process=False,
+        adaptive_class=None,
+    ):
+
+        """Constructs a new DaskSLURMBackend object
+        Parameters
+        ----------
```
Comment (SimonBoothroyd) [inline openff/evaluator/backends/dask.py:727]:
```suggestion
        """
```

File: openff/evaluator/backends/dask.py [hunk 3]
```diff
@@ -695,6 +700,87 @@ def _get_cluster_class(self):
...
+            cluster_type="slurm",
+            adaptive_class=adaptive_class,
+        )
+
+
+    def _get_extra_cluster_kwargs(self):
+
+        extra_kwargs = super(DaskSLURMBackend, self)._get_extra_cluster_kwargs()
+
+        return extra_kwargs
```
Comment (SimonBoothroyd) [inline openff/evaluator/backends/dask.py:776]:
```suggestion
```

Comment (SimonBoothroyd) [reply openff/evaluator/backends/dask.py:776]: You don't need this if you aren't adding any extra kwargs 🙂

**Review by SimonBoothroyd — CHANGES_REQUESTED** [review]
Thanks for this @pavankum ! It would be good if you could undo the changes to the notebooks to keep this PR as single purpose as possible.

Could you also please add the Slurm backend to the test in `tests/test_backends/test_dask.py` on line 27?

File: openff/evaluator/backends/dask.py (hunk 1 above)
Comment (pavankum) [reply openff/evaluator/backends/dask.py:104]: I have both in the env variables on my cluster but slurm manual suggests the newer one

**Comment by SimonBoothroyd** [comment]
The test failures are due to license issues when running the CI from a fork so this should be good to go! Thanks for this @pavankum

**Pull request merged**